internal/pipeline/evaluator_test.go

```diff
@@ -193,3 +207,142 @@ func TestEvaluator_createWorkDir(t *testing.T) {
 		})
 	}
 }
+
+func TestNewPipelineEvaluator(t *testing.T) {
+	type args struct {
+		ctx        context.Context
+		fpath      string
+		policyRepo PolicyRepo
+		namespace  string
+	}
+	tests := []struct {
+		name         string
+		args         args
+		want         *Evaluator
+		wantErr      bool
+		errMsg       error
+		failurePoint string
+	}{
+		{
+			name: "Returns evaluator with appropriate inputs",
+			args: args{
+				ctx:   context.Background(),
+				fpath: "/tmp/pipeline.json",
+				policyRepo: PolicyRepo{
+					PolicyDir: "policy",
+					RepoURL:   "https://example.com/user/foo.git",
+					RepoRef:   "main",
+				},
+				namespace: "pipeline.main",
+			},
+			want: &Evaluator{
+				Context: context.Background(),
+				Target: &DefinitionFile{
+					fpath: "/tmp/pipeline.json",
+					name:  "",
+				},
+				PolicySources: []PolicySource{&PolicyRepo{
+					PolicyDir: "policy",
+					RepoURL:   "https://example.com/user/foo.git",
+					RepoRef:   "main",
+				}},
+				Paths: ConfigurationPaths{
+					PolicyPaths: []string{"/tmp/ec-work-1234/policy"},
+					DataPaths:   []string{"/tmp/ec-work-1234/data"},
+				},
+				TestRunner: runner.TestRunner{
+					Policy:    []string{"/tmp/ec-work-1234/policy"},
+					Data:      []string{"/tmp/ec-work-1234/data"},
+					Namespace: []string{"pipeline.main"},
+					NoFail:    true,
+					Output:    "json",
+				},
+				Namespace:    []string{"pipeline.main"},
+				OutputFormat: "json",
+				workDir:      "/tmp/ec-work-1234",
```

Comment:
Aah, I guess it would complicate the test here...

Reply:
Yes.